src/System.Runtime.Extensions/tests/Performance/Perf.Path.cs

```diff
@@ -62,5 +62,169 @@ public void GetDirectoryName()
+                    {
+                        Path.ChangeExtension(testPath, extension); Path.ChangeExtension(testPath, extension); Path.ChangeExtension(testPath, extension);
+                        Path.ChangeExtension(testPath, extension); Path.ChangeExtension(testPath, extension); Path.ChangeExtension(testPath, extension);
+                        Path.ChangeExtension(testPath, extension); Path.ChangeExtension(testPath, extension); Path.ChangeExtension(testPath, extension);
+                    }
+        }
+
+        [Benchmark]
+        [InlineData(10000)]
+        [InlineData(20000)]
+        [InlineData(30000)]
+        public void GetExtension(int innerIterations)
+        {
+            PerfUtils utils = new PerfUtils();
+            string testPath = utils.GetTestFilePath();
+            foreach (var iteration in Benchmark.Iterations)
+                using (iteration.StartMeasurement())
+                    for (int i = 0; i < innerIterations; i++)
+                    {
+                        Path.GetExtension(testPath); Path.GetExtension(testPath); Path.GetExtension(testPath);
+                        Path.GetExtension(testPath); Path.GetExtension(testPath); Path.GetExtension(testPath);
+                        Path.GetExtension(testPath); Path.GetExtension(testPath); Path.GetExtension(testPath);
+                    }
+        }
+
+        [Benchmark]
+        [InlineData(10000)]
+        [InlineData(20000)]
+        [InlineData(30000)]
+        public void GetFileNameWithoutExtension(int innerIterations)
+        {
+            PerfUtils utils = new PerfUtils();
+            string testPath = utils.GetTestFilePath();
+            foreach (var iteration in Benchmark.Iterations)
+                using (iteration.StartMeasurement())
+                    for (int i = 0; i < innerIterations; i++)
+                    {
+                        Path.GetFileNameWithoutExtension(testPath); Path.GetFileNameWithoutExtension(testPath); Path.GetFileNameWithoutExtension(testPath);
+                        Path.GetFileNameWithoutExtension(testPath); Path.GetFileNameWithoutExtension(testPath); Path.GetFileNameWithoutExtension(testPath);
+                        Path.GetFileNameWithoutExtension(testPath); Path.GetFileNameWithoutExtension(testPath); Path.GetFileNameWithoutExtension(testPath);
+                    }
+        }
+
+        [Benchmark]
+        [InlineData(10000)]
+        [InlineData(20000)]
+        [InlineData(30000)]
+        public void GetFullPath(int innerIterations)
+        {
+            PerfUtils utils = new PerfUtils();
+            string testPath = utils.GetTestFilePath();
+            foreach (var iteration in Benchmark.Iterations)
+                using (iteration.StartMeasurement())
+                    for (int i = 0; i < innerIterations; i++)
+                    {
+                        Path.GetFullPath(testPath); Path.GetFullPath(testPath); Path.GetFullPath(testPath);
```

Comment:
It is radically different for path lengths. We should be testing paths that are < 250,  == 500, and == 1000 characters to keep track of perf (and work on all systems, as some top out at 1024). Paths don't have to physically exist on the system (and shouldn't for the purposes of path manipulation tests).